media-proxy/src/mesh/st2110.cc

```diff
@@ -0,0 +1,153 @@
+#include "st2110.h"
+
+namespace mesh::connection {
+
+st_frame_fmt ST2110::mesh_video_format_to_st_format(int fmt)
+{
+    switch (fmt) {
+    case MESH_VIDEO_PIXEL_FORMAT_NV12:
+        return ST_FRAME_FMT_YUV420CUSTOM8;
+    case MESH_VIDEO_PIXEL_FORMAT_YUV422P:
+        return ST_FRAME_FMT_YUV422PLANAR8;
+    case MESH_VIDEO_PIXEL_FORMAT_YUV422P10LE:
+        return ST_FRAME_FMT_YUV422PLANAR10LE;
+    case MESH_VIDEO_PIXEL_FORMAT_YUV444P10LE:
+        return ST_FRAME_FMT_YUV444PLANAR10LE;
+    case MESH_VIDEO_PIXEL_FORMAT_RGB8:
+        return ST_FRAME_FMT_RGB8;
+    default:
+        return ST_FRAME_FMT_YUV422PLANAR10LE;
+    }
+}
+
+st30_fmt ST2110::mesh_audio_format_to_st_format(int fmt)
+{
+    switch (fmt) {
+    case MESH_AUDIO_FORMAT_PCM_S8:
+        return ST30_FMT_PCM8;
+    case MESH_AUDIO_FORMAT_PCM_S16BE:
+        return ST30_FMT_PCM16;
+    case MESH_AUDIO_FORMAT_PCM_S24BE:
+        return ST30_FMT_PCM24;
+    default:
+        return ST30_FMT_MAX;
+    }
+}
+
+st30_sampling ST2110::mesh_audio_sampling_to_st_sampling(int sampling)
+{
+    switch (sampling) {
+    case MESH_AUDIO_SAMPLE_RATE_48000:
+        return ST30_SAMPLING_48K;
+    case MESH_AUDIO_SAMPLE_RATE_96000:
+        return ST30_SAMPLING_96K;
+    case MESH_AUDIO_SAMPLE_RATE_44100:
+        return ST31_SAMPLING_44K;
+    default:
+        return ST30_SAMPLING_MAX;
+    }
+}
+
+st30_ptime ST2110::mesh_audio_ptime_to_st_ptime(int ptime)
+{
+    switch (ptime) {
+    case MESH_AUDIO_PACKET_TIME_1MS:
+        return ST30_PTIME_1MS;
+    case MESH_AUDIO_PACKET_TIME_125US:
+        return ST30_PTIME_125US;
+    case MESH_AUDIO_PACKET_TIME_250US:
+        return ST30_PTIME_250US;
+    case MESH_AUDIO_PACKET_TIME_333US:
+        return ST30_PTIME_333US;
+    case MESH_AUDIO_PACKET_TIME_4MS:
+        return ST30_PTIME_4MS;
+    case MESH_AUDIO_PACKET_TIME_80US:
+        return ST31_PTIME_80US;
+    case MESH_AUDIO_PACKET_TIME_1_09MS:
+        return ST31_PTIME_1_09MS;
+    case MESH_AUDIO_PACKET_TIME_0_14MS:
+        return ST31_PTIME_0_14MS;
+    case MESH_AUDIO_PACKET_TIME_0_09MS:
+        return ST31_PTIME_0_09MS;
+    default:
+        return ST30_PTIME_MAX;
+    }
+}
+
+void *ST2110::get_frame_data_ptr(st_frame *src) { return src->addr[0]; }
+void *ST2110::get_frame_data_ptr(st30_frame *src) { return src->addr; }
```

Comment:
Do not inline these in `.cc` files. Expand to a normal function block OR move to the header file.

Reply:
Are you sure `src` is always non NULL?

Reply:
Yes, I am sure ```src``` is always non NULL
Fixed in f4af317bd551434672e09b15a6e4ef1fcdd526cc